packages/paste-website/src/pages/primitives/tabs-primitive/index.mdx

```diff
@@ -0,0 +1,471 @@
+/>
+
+---
+
+<contentwrapper>
+
+<PageAside data={props.data.mdx} />
+
+<content>
+
+## Guidelines
+
+### About the Tabs Primitive
+
+The Tabs primitive is an unstyled, functional version of a tabset component. It can be used to build a component following the [WAI-ARIA Tabs Pattern](https://www.w3.org/TR/wai-aria-practices/#tabpanel). Our Tabs component is built on top of this primitive.
+
+This unstyled primitive is to be used when the styled [Tabs](/components/tabs) provided by Paste doesn&rsquo;t meet the requirements needed to solve a unique customer problem. At that point, you are welcome to fallback to this functional primitive to roll your own styled Tabs while still providing a functional and accessible experience to your customers.
+
+This primitive should be used to compose all custom Tabs to ensure accessibility and upgrade paths.
+
+<Callout variant="warning">
+  <CalloutTitle as="h3">Before you roll your own tabset...</CalloutTitle>
+  <CalloutText>
+    We strongly suggest that all components built on top of this primitive get reviewed by the Design Systems team and
+    go through the UX Review process to ensure an excellent experience for our customers.
+  </CalloutText>
+</Callout>
+
+### Examples
+
+#### Horizontal Tabset
+
+<LivePreview
+  scope={{
+    useTabPrimitiveState,
+    TabPrimitiveList,
+    TabPrimitive,
+    TabPrimitivePanel,
+  }}
+  noInline
+>
+  {horizontalExample}
+</LivePreview>
+
+#### Vertical Tabset
+
+<LivePreview
+  scope={{
+    useTabPrimitiveState,
+    TabPrimitiveList,
+    TabPrimitive,
+    TabPrimitivePanel,
+    Stack,
+    Box,
+  }}
+  noInline
+>
+  {verticalExample}
+</LivePreview>
+
+---
+
+### Styling a Custom Tabset
+
+The tab primitive can be styled using Paste components and tokens. By using the `as` prop, we can
+change the underlying element and combine it with another component. In the example below we're combining
+the `TabPrimitive` with a custom `Box` component. We are rendering the `TabPrimitive` as `CustomTab`, passing down
+the necessary props from the state hook and combining those with `Box` styling props.
+
+<LivePreview
+  scope={{
+    useTabPrimitiveState,
+    TabPrimitiveList,
+    TabPrimitive,
+    TabPrimitivePanel,
+    Stack,
+    Box,
+  }}
+  noInline
+>
+  {styledExample}
+</LivePreview>
+
+---
+
+## Usage Guide
+
+This package is a wrapper around the [Reakit Tab](https://reakit.io/docs/tab/). If you&rsquo;re wondering why we wrapped that package into our own, we reasoned that it would be best for our consumers&rsquo; developer experience. For example:
+
+- If we want to migrate the underlying nuts and bolts in the future, Twilio products that depend on this primitive would need to replace all occurrences of `import … from 'x-package'` to `import … from '@some-new/package'`. By wrapping it in `@twilio-paste/x-primitive`, this refactor can be avoided. The only change would be a version bump in the ‘package.json` file for the primitive.
```

Comment:
You might consider re-wording this first bullet. In reading only the first sentence, it seems like I might need to replace all occurrences of that import line. It only becomes clear in the second sentence that I _won't_ have to do that because we wrapped Reakit. I think just switching the order of the sentences and tweaking the wording accordingly would make it more clear.